### DISCUSSION TITLE
Allow clients to work through outages / Enable offline mode

### DISCUSSION THREAD

**Discussion by virtuallyvlad** [standardnotes/forum#2864] [Suggestions]
Can we set up some way to allow the desktop, web, and mobile clients to continue working through system outages?

It seems that with certain system or server issues, [like the one we are encountering today](https://twitter.com/StandardNotes/status/1446103435485581312), the SN clients become unusable.  In this case it is in the form of an endless "Authentication Required" popup.  But historically speaking, it has always been pretty annoying to use the clients during outages.

For this specific case, a solution could be to have that popup only show up once, or once per hour, instead of every 15 seconds.  For a more general solution, perhaps there could be a "Pause Sync" button.  Or it could be some other solution, just so long as the app is not unusable and the user is not continuously bothered during an outage that may last hours.

**Comment by atmoio** [comment]
Typically outages are related to sync, and those shouldn't be annoying, beyond just displaying red error status in the top right. Today we're separating sync infrastructure from authentication, so it's a new class of issues we're seeing. We'll have to look into how that might be handled more peacefully in the client.

**Comment by virtuallyvlad** [comment]
The typical sync-based outages are annoying because **_the bold red error status in the top right will constantly alternate_** between "Saving" and "Sync Unreachable" every few seconds, making the client massively distracting to use for writing.  That could be improved by having it not alternate maybe, and instead just remain on the "Sync Unreachable" message, for example.

The typical sync-based outages also have a tendency of **_producing conflicting copies_**, which are a pain to manually compare and resolve.  I think ideally the user shouldn't see conflicting copies (for example, do you get conflicting copies when using Apple Notes via iCloud?), but if they are unavoidable then perhaps some easier way to compare/sync/resolve them.

Related:
https://github.com/standardnotes/forum/issues/750
https://github.com/standardnotes/forum/issues/1234
https://github.com/standardnotes/desktop/issues/672

**Comment by dddvision** [comment]
I am getting a proliferation of "Conflicted Copy" near-duplicate messages today on macos 3.8.21. While writing a few sentences, I can end up with several "Conflicted Copy" messages. This is a severe problem, and it is the only reason why I did not purchase a 5-year subscription today.

**Comment by myreli** [comment]
I believe the sync-related messages have been solved through the updated UI, which now displays an optional small indicator of the syncing status. 

It seems like this issue is unlikely to receive more interaction, so I'll close it. 

_(This is a community effort to triage older issues in the forum. If this was a mistake please feel free to re-open the issue.)_

**Comment by virtuallyvlad** [comment]
I just tested it and the current behavior (v3.149.5) is a spinning refresh indicator followed by a red (!) icon.  Since this would occur after every keystroke during an outage, it would still probably be too annoying to use comfortably.  But in the interest of letting go of old issues, I'll agree to close this.  Hopefully the sync indicator can be improved further in future versions to be even more subtle.  Whether it's a blinking green light or a blinking red light, it's an annoyance that's unique to Standard Notes and should be handled more elegantly.


> an optional small indicator of the syncing status.

It can be hidden?  Or how optional?

**Comment by myreli** [comment]
Hi @virtuallyvlad,

Yes, you can toggle under `Settings > General > Tools Show note saving status while editing`. 

However I just noticed that _"Error statuses are always shown regardless of preference."_. So you are correct. Even if it's much less disruptive, the red indicator remains constant when completely offline. I personally haven't been bothered by it yet, but agree it could have a more elegant solution. 

Perhaps a monochrome error indicator would be better, or even the option to disable it completely. How you imagine this experience?

**Comment by virtuallyvlad** [comment]
The issue is that movement and bright colors distract from writing.  So you could reduce one or both.  The most subtle would be a stable monochrome indicator, and the least subtle would be what we have now: a moving indicator that also has a bright color.

I like the way it works with that toggle disabled (i.e., hiding the note saving status while editing), except the blinking red (!) icon should be either a stable red (!) icon or the stable (i) icon.
